app/src/js/components/Collections/add.js

```diff
@@ -1,40 +1,67 @@
 'use strict';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { withRouter } from 'react-router-dom';
-import { createCollection } from '../../actions';
+import { createCollection, getSchema } from '../../actions';
 import { getCollectionId, collectionHref } from '../../utils/format';
+import { removeReadOnly } from '../FormSchema/schema';
 import AddRaw from '../AddRaw/add-raw';
 
-const getBaseRoute = function (collectionId) {
-  if (collectionId) {
-    return collectionHref(collectionId);
-  } else {
-    return '/collections/all';
-  }
-};
+const AddCollection = ({ location = {}, collections, dispatch, schema }) => {
+  const [defaultValue, setDefaultValue] = useState({});
+  const { state: locationState } = location;
+  const { name, version } = locationState || {};
+  const collectionId = getCollectionId({name, version});
+  const { collection: collectionSchema } = schema || {};
+  const { map: collectionsMap } = collections || {};
+  const isCopy = !!(name && version);
+
+  useEffect(() => {
+    if (isCopy) {
+      dispatch(getSchema('collection'));
+    }
+  }, [isCopy, dispatch]);
+
+  useEffect(() => {
+    const record = collectionsMap[collectionId];
+    const { data } = record || {};
+    if (isCopy && data && collectionSchema) {
+      setDefaultValue(removeReadOnly(data, collectionSchema));
+    }
+  }, [collectionSchema, collectionId, collectionsMap, isCopy]);
 
-class AddCollection extends React.Component {
-  render () {
-    return (
-      <AddRaw
-        pk={'new-collection'}
-        title={'Add a collection'}
-        primaryProperty={'name'}
-        state={this.props.collections}
-        createRecord={createCollection}
-        getBaseRoute={getBaseRoute}
-        getPk={getCollectionId}
-      />
-    );
-  }
-}
+  const getBaseRoute = (pk) => {
+    let id = pk || collectionId;
+    if (id && id !== 'unknown') {
+      return collectionHref(id);
+    } else {
+      return '/collections';
+    }
+  };
```

Comment:
Ignorable suggestion, but here's a way to simplify the function above:
```
const getBaseRoute = (pk = collectionId) =>
  (pk && pk !== 'unknown') ? collectionHref(pk) : '/collections';
```

Reply:
Good call, this is so much cleaner! Changing.